### PR TITLE
Allow for a different default search provider for private tabs

### LINKED ISSUE #11305: Allow user to opt into a different search engine for private tabs
### Description
The private tab screen should show a recommended search engine, which would apply only to private tabs if enabled.  It would go into the place where the red selection is in the screenshot below:
screen shot 2017-10-05 at 11 10 20 am

The engine to use (and text to be displayed) is still TBD, but this could look something like this:
screen shot 2017-10-05 at 11 24 59 am

Since we don't have a way to do private windows, simply showing the toggle in the private tab should work great.  

In the future, we may allow the search screen to show (and allow changing) the search engine used for private tabs (see https://github.com/brave/browser-laptop/issues/11218)

### Open items
- [ ] Search engine to use
- [ ] Resources to use (SVG, icons, etc)
- [ ] Text to use / final mock up

### PR DESCRIPTION
**Pull request by petemill**
Introduces setting: 'search.use-alternate-private-search-engine which defines whether or not to use a different search provider (currently hardcoded to DuckDuckGo) when searching from a Private Tab.

Uses this setting when searching in the UrlBar when a private frame is active.
Setting can be changed from about:preferences or about:newtab (in a Private Tab)

Fix #11305

**This PR has 3 commits** since it also introduces other semi-related, but separatable changes:
1. New fonts, used by this design but intended to be used in other features in the future. These fonts have been downloaded for offline use from Google Fonts using a script that I made. The script makes sure that the fonts are downloaded in an optimal format for quality and performance, yet allowing flexibility with weights and unicode characters. Both fonts are licensed under the SIL Open Font License.
2. about:new page typographic styles - mainly using the new fonts and associated style rules.
3. Private Search which includes:
   - Private Search urlBar functionality
   - Preferences setting for adjusting Private Search
   - about:new for Private Tabs containing
     - New fonts and style rules as per the non-private version
     - The same setting from Preferences to toggle Private Search
     - Text and Branding for Private Search

They could be split in to 3 PRs, but I've left them here for now which may be preferable since they will all require merging to multiple release branches, when approved, due to interdependencies.

# Test Plan:

- Unit tests included

### Style updates
Do the new tab page and private new tab page render correctly, with no errors?

### Private Search
Without changing any settings (since Private Search should be off by default):
- search in a non-private tab should use default search provider (google)
- search in a private tab should use default search provider (google)

With the setting changed from the private New Tab page, or the Tab section of the Preferences page:
- search in a non-private tab should use default search provider (google)
- search in a private tab should use the hardcoded Private Search provider (duckduckgo)

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Reviewer Checklist:

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: js/about/newprivatetab.js [hunk 1]
```diff
@@ -29,6 +38,17 @@ class NewPrivateTab extends React.Component {
           <p className={css(styles.text)} data-l10n-id='privateTabText1' />
           <p className={css(styles.text)} data-l10n-id='privateTabText2' />
           <p className={css(styles.text, styles.italic)} data-l10n-id='privateTabText3' />
+          <h2 className={css(styles.sectionTitle)} data-l10n-id='privateTabSearchSectionTitle' />
```
Comment (luixxiul) [inline js/about/newprivatetab.js]: FYI there is a component for section titles (please see: `about:styles#sectionTitle`)

**Comment by petemill** [comment]
After a lengthy #product discussion and call, the design of this has been changed. I'll implement the new layout and then submit for further product feedback...

File: js/about/newTabComponents/clock.js [hunk 2]
```diff
@@ -11,14 +11,26 @@ class Clock extends React.Component {
     this.state = this.getClockState(new Date())
   }
   get formattedTime () {
-    const time = this.state.currentTime
-    return time.replace(' AM', '').replace(' PM', '')
+    return this.state.currentTime.map(component => {
```
Comment (bsclifton) [inline js/about/newTabComponents/clock.js]: nice cleanup here 😄

Comment (petemill) [reply js/about/newTabComponents/clock.js]: It was mainly so I could vertically center the ':' character in the time. I just wasn't sure if it existed in all locales, so I did this...

**Comment by bsclifton** [comment]
master f522e1b1229e6e42248631aa2b40f69b5792c52d
0.21.x 5ec05d4a25b8cd4707ebd15c1fc78414285069f9
0.20.x 0ee72ab2a2126902573257762463583c7aadba02
0.19.x working through manual merge...

**Comment by petemill** [comment]
@bsclifton is there a process we need to go through to make sure the translations for the new strings added get added in to the milestone branches?

**Comment by bsclifton** [comment]
@petemill as long as the `en-US` strings get updated, we should be good to go.  The process is documented here:
https://github.com/brave/browser-laptop/blob/master/docs/translations.md

We only land new translations in master unfortunately- backporting them is risky and would be a manual process

**Pull request merged**